tests/baselines/reference/importTypeInJSDoc.types

```diff
@@ -37,14 +37,14 @@ export = MyClass;
  */
 
 let a = /** @type {Foo} */(/** @type {*} */(undefined)); 
->a : import("tests/cases/conformance/types/import/externs")
+>a : any

```

Comment:
? There's a typedef for `Foo` right above this - so why is it `any` now?

Reply:
Previously we were incorrectly treating `/** @type {Foo} */` as a type annotation for the `let a` rather than as a cast. Now it looks like `getTypeForVariableLikeDeclaration` doesn't get a type from `tryGetTypeFromEffectiveTypeNode` (which it shouldn't), but then escapes early returning `autoType` because `isNullOrUndefined` skips parentheses (ignoring type casts). Quickinfo shows `any` at the declaration site, but shows the correct type in any position immediately afterwards.

I can augment `isNullOrUndefined` to check whether any of the parenthesis are JSDoc type casts.

Reply:
Yeah, that's sounds right. I image we probably shouldn't ignore casts.